### PR TITLE
add newton step convergence checks to steadystate solver

### PR DESCRIPTION
**Pull request by FFroehlich**

### CODE REVIEW

File: include/amici/solver.h [hunk 1]
```diff
@@ -867,6 +867,40 @@ class Solver {
...
+     * @brief Indicates whether convergence checks for steadystate computation are performed on the
+     * newton step (true) or directly on the right hand side (false).
+     * @return boolean flag
+     */
+    bool getNewtonStepSteadyStateCheck() const {
+        return newton_step_steadystate_conv_;
+    }
+    
+    /**
+     * @brief Indicates whether convergence checks for are also performed for sensitivity equations (true)
```
Comment (dweindl) [inline include/amici/solver.h:881]:
```suggestion
     * @brief Indicates whether convergence checks for steadystate computation are also performed for sensitivity equations (true)
```
?

File: include/amici/solver.h [hunk 2]
```diff
@@ -867,6 +867,40 @@ class Solver {
...
+    /**
+     * @brief Sets whether convergence checks for steadystate computation are performed on the
+     * newton step (true) or directly on the right hand side (false, default).
+     */
+    void setNewtonStepSteadyStateCheck(bool flag) {
+        newton_step_steadystate_conv_ = flag;
+    }
+    
+    /**
+     * @brief Sets whether convergence checks for are also performed for sensitivity equations
```
Comment (dweindl) [inline include/amici/solver.h:898]:
```suggestion
     * @brief Sets whether convergence checks for steadystate computation are also performed for sensitivity equations
```

File: src/hdf5.cpp [hunk 3]
```diff
@@ -736,6 +736,14 @@ void writeSolverSettingsToHDF5(Solver const& solver,
     ibuffer = static_cast<int>(solver.getReturnDataReportingMode());
     H5LTset_attribute_int(file.getId(), hdf5Location.c_str(),
                           "rdrm", &ibuffer, 1);
+    
+    ibuffer = static_cast<int>(solver.getNewtonStepSteadyStateCheck());
+    H5LTset_attribute_int(file.getId(), hdf5Location.c_str(),
+                          "newton_step_steadystate_conv_", &ibuffer, 1);
```
Comment (dweindl) [inline src/hdf5.cpp:742]: Do we want to keep the trailing underscore in the attribute name? No strong opinion.

File: src/steadystateproblem.cpp [hunk 4]
```diff
@@ -435,22 +439,39 @@ realtype SteadystateProblem::getWrms(Model *model,
            to zero at all. So we need xQBdot, hence compute xQB first. */
         computeQBfromQ(model, xQ_, xQB_);
         computeQBfromQ(model, xB_, xQBdot_);
+        if (newton_step_conv_)
+            throw NewtonFailure(
+                AMICI_NOT_IMPLEMENTED,
+                "Newton type convergence check is not impelemtend for adjoint "
```
Comment (dweindl) [inline src/steadystateproblem.cpp:445]:
```suggestion
                "Newton type convergence check is not implemented for adjoint "
```

**Review by dweindl — APPROVED** [review]
:+1: 
Looks good, as far as I can tell

File: src/hdf5.cpp (hunk 3 above)
Comment (FFroehlich) [reply src/hdf5.cpp:742]: :+1: no, probably not

**Pull request merged**